amundsen_application/static/js/components/DashboardPage/ImagePreview/index.tsx

```diff
@@ -1,76 +1,64 @@
 import * as React from 'react';
-import { bindActionCreators } from 'redux';
-import { connect } from 'react-redux';
 
-import { getDashboardPreview } from 'ducks/dashboard/reducer';
-import { GetDashboardPreviewRequest } from 'ducks/dashboard/types';
-import { GlobalState } from 'ducks/rootReducer';
+import Linkify from 'react-linkify'
 
 import LoadingSpinner from 'components/common/LoadingSpinner';
 
 import * as Constants from './constants';
 import './styles.scss';
 
-export interface OwnProps {
+export interface ImagePreviewProps {
   uri: string;
+  dashboard_url?: string; // TODO ttannis: Pass dashboard url after picking up connect_dashboard changes
 }
 
-export interface StateFromProps {
-  url: string;
-  errorMessage: string;
-  errorCode: number | undefined;
+interface ImagePreviewState {
   isLoading: boolean;
+  hasError: boolean;
 }
 
-export interface DispatchFromProps {
-  getPreviewImage: (payload: { uri: string }) => GetDashboardPreviewRequest
-}
-
-export type ImagePreviewProps = StateFromProps & DispatchFromProps & OwnProps;
-
-export class ImagePreview extends React.Component<ImagePreviewProps, {}> {
+export class ImagePreview extends React.Component<ImagePreviewProps, ImagePreviewState> {
   constructor(props) {
     super(props);
+
+    this.state = {
+      isLoading: true,
+      hasError: false,
+    }
   }
 
-  componentDidMount = () =>  {
-    this.props.getPreviewImage({ uri: this.props.uri });
+  onSuccess = () => {
+    this.setState({ isLoading: false, hasError: false });
   }
 
-  render = () =>  {
-    const { errorCode, errorMessage, isLoading, url } = this.props;
-    let content;
-    if (isLoading) {
-      content = <LoadingSpinner/>;
-    }
-    else if (errorCode) {
-      content =  <div className="loading-error-text body-placeholder">{ errorMessage }</div>;
-    }
-    else {
-      content = <img src={this.props.url} height="auto" width="100%" />;
-    }
+  onError = (e: React.SyntheticEvent<HTMLImageElement>) => {
+    this.setState({ isLoading: false, hasError: true });
```

Comment:
Would it be possible to check the error code from `e`?

Reply:
No we don't get any network information from these kind of events, just some of the usual DOM related things.